File: server/FileController.java
package com.vaticle.web.main.server;

import play.mvc.Controller;
import play.mvc.Result;

import java.nio.file.Path;
import java.nio.file.Paths;

public class FileController extends Controller {
    private Path basedir;
    private String robotsTxt;

    public FileController(Path basedir, String robotsTxt) {
        this.basedir = basedir;
        this.robotsTxt = robotsTxt;
    }

    public Result serve(String file, String defaultFile) {
        Path path = basedir.resolve(file);
        if (path.toFile().isFile()) {
            return ok(path);
        } else if (path.resolve(defaultFile).toFile().isFile()) {
            return ok(path.resolve(defaultFile));
        } else if (Paths.get(path.toString() + ".html").toFile().isFile()) {
            return ok(Paths.get(path.toString() + ".html"));
        } else {
            return ok(basedir.resolve(defaultFile));
        }
    }

    public Result robotsTxt() {
        return ok(robotsTxt);
    }
}

File: server/FileController.kt
package grakn.web

import play.mvc.Controller
import java.nio.file.Path

class FileController(private val basedir: Path) : Controller() {

    fun serve(file: String, defaultFile: String): play.mvc.Result {
        val path: Path = basedir.resolve(if (file.isNotEmpty()) file else defaultFile)
        return if (path.toFile().exists()) {
            ok(path)
        } else {
            ok(basedir.resolve(defaultFile))
        }
    }
}
